control_plane/src/storage.rs

```diff
@@ -317,20 +330,17 @@ impl PageServerNode {
 
         // Wait until process is gone
         for i in 0..600 {
-            let signal = None; // Send no signal, just get the error code
-            match kill(pid, signal) {
-                Ok(_) => (), // Process exists, keep waiting
-                Err(Errno::ESRCH) => {
+            match self.check_status() {
+                Ok(()) => (), // Process exists, keep waiting
+                Err(_) => {
+                    std::fs::remove_file(&pid_file).with_context(|| {
```

Comment:
race condition: if the pageserver exited, and a new one started already, this will remove the pid file of the new instance

Reply:
This one is quite hard to detect, but I wonder, how did we work with this set-up before? 
We usually start pageservers in separate directories per test and generally did not test `neon_base` for disaster scenarios, neither should we at the current moment.

So, might be just ok for now? I'm not sure I want to create any pid files from pageserver, since its deployment does not care about them at all, preferring systemd (k8s later) to handle such cases.

Reply:
I don't want to rely on systemd or kubernetes for this. If you accidentally launch two pageservers on the same data directory, you can get very serious and hard-to-detect corruption. I really want to have a safety mechanism in pageserver itself to prevent that.

Even a PID file isn't 100% bulletproof against every scenario, like if you mounted the same data directory from two different hosts over NFS and ran pageserver in both of them. But accidentally launching two pageservers on the same host is a very easy mistake to make, if you ssh into the server and fiddle around manually, for example.

Reply:
> This one is quite hard to detect, but I wonder, how did we work with this set-up before?

`daemonize` uses `flock(LOCK_EX)` on the pid file to ensure that only one process is running.  Could we continue to use `daemonize`, without the forking? Or write similar open+flock code by hand.

Reply:
Went with the lockfile approach, since safekeeper had it already.
Also put the pidfile inside of it.